pkg/reconciler/taskrun/resources/taskref.go

```diff
@@ -123,6 +123,15 @@ func GetTaskFunc(ctx context.Context, k8s kubernetes.Interface, tekton clientset
 	}
 }
 
+// GetStepActionFunc is a factory function that will use the given Ref as context to return a valid GetStepAction function.
+func GetStepActionFunc(tekton clientset.Interface, namespace string) GetStepAction {
```

Comment:
do we consider having a new file to store stepaction related functions? Not something we need to do right now

Reply:
May be. Lets see if the future PRs need further refactoring.